test/DataLayouts/unit_mapreduce.jl

```diff
@@ -0,0 +1,117 @@
+#=
+julia --project
+using Revise; include(joinpath("test", "DataLayouts", "unit_mapreduce.jl"))
+=#
+using Test
+using ClimaCore.DataLayouts
+import ClimaComms
+import Random
+ClimaComms.@import_required_backends
+
+function test_mapreduce_1!(device, data)
+    Random.seed!(1234)
+    device isa ClimaComms.CUDADevice & return nothing
```

Comment:
It doesn't make much difference, but I would swap the order of the two lines. This is so that readers can immediately see that when `device isa CUDADevice` this function exits immediately. (It's perfectly fine if you prefer it the way you have already)

Reply:
I don't have a strong preference, I'm hoping that we can soon fix this and just test mapreduce on datalayouts.